README.md

```diff
@@ -16,3 +17,11 @@ Also give a theoretical argument for why X could or could not be correct, based
 on the complexity of the general sorting problem we covered in class.
 
 Add your answers to this markdown file.
+
+## Testing for $O(n)$
+
+If we only have a blackbox algorithm, then we'll have to test it and graph it. I would test it with different cases, like being sorted, slightly sorted, randomly shuffled, completely reversed for the worst case, and lists of the same number, and make about 30 different versions of each case, going from 1 to 30 length. I would also test empty lists. Then we can graph the time it took to sort over the change in length and test it against $y = x$ which will be my linear regression line. If the correlation coefficient $r$ is significantly close to 1, then we do truly have a linear algorithm, otherwise the algorithm does not have a $O(n)$ complexity. 
```

Comment:
What about any constant factors or constants?

Reply:
I'm not sure what you mean by that. Did you mean constants in the graph? I think that would make the correlation coefficient worse and prove it's not linear.

Reply:
What if the correlation is of the form $f(x) = ax + b$?

Reply:
I understand what you mean, I think if I found the regression and it was linear and $r$ was significantly close to 1, then it would have a linear complexity right?